### ISSUE TITLE
Loss computations for CLUE

### ISSUE THREAD

**Issue by jmayank23** [gao-lab/GLUE#74]
I was really interested in the CLUE method, so I was looking at the code and read the paper to implement it. I noticed that the KL divergence terms in the paper and the code looked different, so I was wondering if you could help me with this part; if you could share which approach worked better and like some details about how to compute it, that would be really helpful. 
Sharing reference screenshots here-

<img width="676" alt="Screen Shot 2023-06-08 at 12 13 34 PM" src="https://github.com/gao-lab/GLUE/assets/27727185/f12f934f-a573-4c97-81dc-073360b68f97">

![Screen Shot 2023-06-08 at 12 14 26 PM](https://github.com/gao-lab/GLUE/assets/27727185/221502e7-ae55-40f8-a621-24d101e85223)

Thank you

**Comment by Jeff1995** [comment]
Hi @jmayank23. Thanks for your interest in CLUE!

The differences you mentioned are probably the followings:

1. The code normalizes both the negative log likelihood (`.mean()`) and the KL term (`/ x[k].shape[1]`) by the number of features in the data modality, so that different modalities receive more or less the same attention during training.
2. The code has a hyperparameter `self.lam_kl` that controls the weight of the KL term, which was optimized using cross-validation. This was omitted in the paper to steer clear for the main message of cross encoding.

Let me know if there are further problems.
